### ISSUE TITLE
How to push git tags

### ISSUE THREAD

**Issue by dragontho** [nus-cs2103-AY1920S1/forum#11]
Hi prof,
Quick question, but how exactly do you push git tags? After a quick search online I came across this answer:
https://stackoverflow.com/questions/5195859/how-do-you-push-a-tag-to-a-remote-repository-using-git
However, after adding the --follow-tags option to the end of git push, I could not find any evidence that the tag has been pushed. Furthermore, the answer also discourages the use of git push --tags, for which I also still do not know the reason. Are there multiple ways to push tags to the repository and if so which is the most appropriate? Also, how do you find evidence of your tags being pushed? Please redirect to another answer if the issue has been resolved already.

**Comment by damithc** [comment]
Let's see if someone who has pushed tags already can help. In the meantime, this is how you can check if the tags are present in the remote repo:
![image](https://user-images.githubusercontent.com/1673303/63505077-6d411180-c505-11e9-842b-13b966515570.png)

**Comment by dawo5010** [comment]
I followed this tutorial:
https://git-scm.com/book/en/v2/Git-Basics-Tagging

tl;dr for you: `git push origin <tagname>` for a single tag, `git push origin --tags` to upload all tags not already present remotely.

**Comment by uggi121** [comment]
In the command-line, type : -
    git ls-remote --tags origin

You should see a list of pushed tags

**Comment by CarbonGrid** [comment]
You probably only need to understand the following:
git tag
git tag -d <tagname>
git tag -f <tagname>
git push --tags

**Comment by TheRealRavager** [comment]
The difference between the two commands is that `git push --follow-tags` only pushes tags that are both annotated (similar idea to commit message) and reachable (in the current commit history).

Let's start with reachability. Imo it doesn't make sense to have a tag pointing to an unreachable commit (unless you, for some reason, want to keep a copy of it) and more importantly may confuse people.

As for annotations, they are there to give meaning to the tag you have created by giving a short summary of what the tag is for (e.g. what does "Level-6" of your project mean? What major change does it introduce?).

Hope this helped to clarify some of your doubts. I've only recently read up on tags so please correct me if I'm wrong. The Stack Overflow post links to another post that explains annotated tags and their use cases quite well, I recommend giving it a read.

https://stackoverflow.com/questions/11514075/what-is-the-difference-between-an-annotated-and-unannotated-tag

**Comment by dragontho** [comment]
Thanks all for the help!

**Comment by damithc** [comment]
Thanks everyone who pitched in to help 👍
